Create STYLEGUIDE.md

oddlittlebird opened the pull request:
Fixes #19134

oddlittlebird committed:
Create STYLEGUIDE.md

oddlittlebird committed:
Update STYLEGUIDE.md

Added a placeholder for image guidelines, someone needs to add specificity

Added Capitalization section

Expanded Word usage

oddlittlebird committed:
Update STYLEGUIDE.md

Edited panel definition

bharjala reviewed (changes requested):
Two initial suggestions:
- Add a link to [Chicago Manual of Style](https://www.chicagomanualofstyle.org/home.html)
- Add an example for active / passive voice, similar to your example for imperative second person.
- - Add an example of sentence case, headings that start with verbs, etc. 

For new tech writers and contributors, clear, upfront examples are helpful.

marcusolsson commented:
I've been looking at the current images:

Preferred format: PNG
DPI: Most images are 72 DPI with some outliers

Width range: 32px – 3712px (most between 430px – 1290px)
Height range: 30px – 1854px (most between 30px – 910px)

Biggest one seems to be: https://github.com/grafana/website/blob/master/static/static/img/docs/v60/explore_split.png

For now, maybe set maximum size to 3840×2160 (4K), for high-quality screenshots?

As for the minimum size, I'm thinking 256x256. There are a handful of images smaller than that, but I'm debating whether that's too small. Here's an example: https://github.com/grafana/website/blob/master/static/static/img/docs/v60/explore_menu.png

oddlittlebird committed:
Update STYLEGUIDE.md

Applied Brenda and Marcus's edits

oddlittlebird commented:
Applied edits.

We still lack information for how users should deal with image callouts.

I think we need to give guidelines, tell them to just take the screenshot and we will add callouts, or tell them Grafana personnel will supply updated images.

marcusolsson commented on docs/STYLEGUIDE.md:
```diff
@@ -0,0 +1,65 @@
+* Use active voice. Avoid passive voice.
+  - Passive: The zombie was killed.
+  - Active: I killed the zombie.
+* Write in the imperative second person. Examples: You can do a thing. Click the panel. Shoot the zombie.
```
I'd like to use a friendlier example than _shoot the zombie_, e.g. eat the cake, pet the dog.

marcusolsson commented on docs/STYLEGUIDE.md:
```diff
@@ -0,0 +1,65 @@
+  - Not: The panel will open.
+  - Use: The panel opens.
+    or
+    Grafana opens the panel.
```
Maybe put this on same line:

- Not: The panel will open.
- Use: The panel opens, or, Grafana opens the panel.

... or add another Use item:

- Not: The panel will open.
- Use: The panel opens.
- Use: Grafana opens the panel.

marcusolsson commented on docs/STYLEGUIDE.md:
```diff
@@ -0,0 +1,65 @@
+* Preferred DPI is 72.
+* Assume all graphics will be exclusively viewed on the web.
+* Maximum image size is 3840px X 2160px.
+* Screenshots should be readible, but not too large.
```
readible -> readable?

oddlittlebird replied:
ARGH. And that is why we have editors.

marcusolsson commented on docs/STYLEGUIDE.md:
```diff
@@ -0,0 +1,65 @@
+### Capitalization
+
+* Grafana, Loki, and Prometheus are always capitalized unless part of a code block.
+* `amtool` is never capitalized.
```
This is too specific, and not a Grafana term. I think we should remove it, otherwise we'd have to list a lot of these.

marcusolsson commented:
Also, I'm debating whether the style guide should go into https://github.com/grafana/grafana/tree/master/style_guides or whether this should be a proper page under "Contribute" in the documentation.

marcusolsson commented:
I think best place for now is to put it next to the other style guides in the style_guides directory, and link from the docs guide.

oddlittlebird committed:
Moved style guide to style_guides folder and renamed, added README to the style_guides folder, other minor edits

oddlittlebird committed:
Update doc-style-guide.md

marcusolsson reviewed (changes requested):
This is great stuff! 

One minor thing before I think we can merge this. None of the other style guides have the `-style-guide` prefix in their filename. I think we should be consistent here:

Basically rename the file to just `docs.md`.

Almost there!

oddlittlebird committed:
Style updates

Changed a couple doc names for consistency, updated file naming conventions and README links

oddlittlebird commented:
I renamed the style guides to match their titles with the exception of themes.md, since that one is the subject of a different issue.

oddlittlebird committed:
Corrected file names

Changed file names back to original, clarified file naming convention in documentation-style-guide

marcusolsson reviewed (approved):
Looks great! Thanks for helping out with this! 👍

oddlittlebird commented:
My pleasure.

On Fri, Sep 20, 2019, 9:45 AM Marcus Olsson <notifications@github.com>
wrote:

> *@marcusolsson* approved this pull request.
>
> Looks great! Thanks for helping out with this! 👍
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/grafana/grafana/pull/19190?email_source=notifications&email_token=AMNF6KJFXBWVVFUPUZZOOF3QKT42DA5CNFSM4IXXLULKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCFOIUAA#pullrequestreview-291277312>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMNF6KJYIUKCY56Z5H7XYULQKT42DANCNFSM4IXXLULA>
> .
>
